### PR TITLE
Write add interview feature for DG

### PR DESCRIPTION
**Pull request by headcube1**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -273,7 +273,53 @@ The following sequence diagram illustrates step 4:
...
+### Add Interview Feature
+#### Implementation
+
+The add interview mechanism is facilitated by `AddInterviewCommand`. They all extend the `Command` with fields called `description`, `applicant` phone number, 
+`interviewer` phone number, `date` of interview, `startTime` as well as `endTime`. An `Interview` is created then added to the list. 
+AddInterviewCommand implements the following operations:
+
+* `AddInterviewCommand#excute()`  —  Adds the encapsulated `Interview` to the list.
+
+The above `excute` operation utilises `ModelManager#updateFilteredPersonList()` implemented from the Model interface to obtain the list of `applicant` and `interviewer` phone numbers.
```
Comment (yashpola) [inline docs/DeveloperGuide.md:286]: Typo in `execute`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -273,7 +273,53 @@ The following sequence diagram illustrates step 4:
...
+This is followed by checking the correctness of the phone numbers before creating an `Interview` object to be added into the interview list. The operation `excute` then utilises
```
Comment (yashpola) [inline docs/DeveloperGuide.md:287]: Perhaps "validity" instead of "correctness" is clearer?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -273,7 +273,53 @@ The following sequence diagram illustrates step 4:
...
+`ModelManager#addInterview()` implemented from the Model to add the `Interview` to the list. The operation `excute` also utilises `ModelManager#sortInterview()` to the `interview` objects by `date`, `startTime` and `endTime`.
```
Comment (yashpola) [inline docs/DeveloperGuide.md:288]: Good on including that `sort` is also performed!

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -273,7 +273,53 @@ The following sequence diagram illustrates step 4:
     * Pros: Greater flexibility in setting statuses.
     * Cons: Possible performance issues when updating multi-statuses, as well as graphical design overhead when choosing how to sort and render multiple statuses.
 
-_{more aspects and alternatives to be added}_
+
+### Add Interview Feature
+#### Implementation
+
+The add interview mechanism is facilitated by `AddInterviewCommand`. They all extend the `Command` with fields called `description`, `applicant` phone number,
```
Comment (yashpola) [inline docs/DeveloperGuide.md:280]: Good elaboration on what information an `AddInterviewCommand` encapsulates

**Review by yashpola — APPROVED** [review]
LGTM, minor nits

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -273,7 +273,53 @@ The following sequence diagram illustrates step 4:
...
+
+Note:
+
+* The command expects all arguments to be filled and will result in `ParseException` indicating invalid command format otherwise.
+* When a duplicate interview is entered, it will result in a `CommandException` indicating a duplicate interview has been entered.
+* When incorrect phone numbers are entered, it will result in a `CommandException` indicating which phone number is incorrect.
+
+The following sequence diagram shows demonstrates step 3:
+
+#### Design considerations:
```
Comment (gingerbreaf) [inline docs/DeveloperGuide.md:310]: Well Explained!

**Review by gingerbreaf — APPROVED** [review]
LGTM

**Pull request merged**